bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VolvoOnCallBridgeHandler.java

```diff
@@ -0,0 +1,154 @@
+        VolvoOnCallBridgeConfiguration configuration = getConfigAs(VolvoOnCallBridgeConfiguration.class);
+
+        try {
+            httpHeader.setProperty("Authorization", configuration.getAuthorization());
+            try {
+                customerAccount = getURL(SERVICE_URL + "customeraccounts/", CustomerAccounts.class);
+                if (customerAccount.username != null) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Incorrect username or password");
+                }
+            } catch (IOException | JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        } catch (UnsupportedEncodingException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("VolvoOnCall Bridge is read-only and does not handle commands");
+    }
+
+    public String[] getVehiclesRelationsURL() {
+        if (customerAccount != null) {
+            return customerAccount.accountVehicleRelationsURL;
+        } else {
+            return new String[0];
+        }
+    }
+
+    public <T> T getURL(Class<T> objectClass, String vin) throws IOException, JsonSyntaxException {
+        String url = SERVICE_URL + "vehicles/" + vin + "/" + objectClass.getSimpleName().toLowerCase();
+        return getURL(url, objectClass);
+    }
+
+    public <T> T getURL(String url, Class<T> objectClass) throws IOException, JsonSyntaxException {
+        String jsonResponse = HttpUtil.executeUrl("GET", url, httpHeader, null, JSON_CONTENT_TYPE, REQUEST_TIMEOUT);
+        return gson.fromJson(jsonResponse, objectClass);
+    }
+
+    public boolean postURL(String URL, @Nullable String body)
+            throws IOException, InterruptedException, JsonSyntaxException {
+        InputStream inputStream = body != null ? new ByteArrayInputStream(body.getBytes(StandardCharsets.UTF_8)) : null;
+        String jsonString = HttpUtil.executeUrl("POST", URL, httpHeader, inputStream, null, REQUEST_TIMEOUT);
+        PostResponse postResponse = gson.fromJson(jsonString, PostResponse.class);
+        if (postResponse.errorLabel == null) {
+            while (postResponse.status == Status.STARTED || postResponse.status == Status.DELIVERED) {
+                Thread.sleep(1000);
+                postResponse = getURL(postResponse.serviceURL, PostResponse.class);
+            }
+            if (postResponse.status == Status.SUCCESSFULL) {
+                logger.debug("Success !");
+                return true;
+            } else {
+                logger.debug("Loop finished with status {}", postResponse.status.toString());
```

Comment:
You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.